src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2768,6 +2760,33 @@ grpc_error *grpc_chttp2_incoming_byte_stream_finished(
   return error;
 }
 
+static void incoming_byte_stream_shutdown(grpc_exec_ctx *exec_ctx,
```

Comment:
Relatedly, doesn't this fail the invariant that the error (whatever it may be) passed to shutdown will be returned to all future calls to pull? Because this shutdown, resets the stream, which will clear the error.